feat(upgrade): allow setting the angularjs lib at runtime

vsavkin opened the pull request:
This PR adds an ability to reset the angularjs library, which is often needed when Angular
is loaded lazily using RequireJS.

**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


**What kind of change does this PR introduce?** (check one with "x")
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** (You can also link to an open issue here)

Currently it is not possible to reset the angular lib at runtime when using ngupgrade. This is problematic for many apps that load angularjs lazily. 

**What is the new behavior?**

Now it is possible.

**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...


**Other information**:

petebacondarwin commented:
I think in principle this is a good change.
I need to sleep on the public API change though.

mhevery commented:
@vsavkin can you rebase to get the PR green again?

vsavkin committed:
feat(upgrade): allow setting the angularjs lib at runtime

This PR adds an ability to reset the angularjs library, which is often needed when Angular
is loaded lazily using RequireJS.

vsavkin commented:
@mhevery done

mhevery commented:
@petebacondarwin please approve so that we can get this merged.

petebacondarwin commented:
@mhevery the pullapprove that is missing is for the `public_api` group, which is you and @IgorMinar. Given that you have approved it I don't undertstand why it is not yet green.

tbosch commented:
Note that hits broke in G3, and I created https://github.com/angular/angular/pull/16000 instead.

petebacondarwin commented:
Sorry about that @tbosch - how can we prevent that from happening in the future?

tbosch commented:
@petebacondarwin So far there is no other way, as you don't have access to G3 :-(
